### ISSUE TITLE
post-install-cmd scripts fail (missing /tmp/app/vendor/autoload.php)

### ISSUE THREAD

**Issue by till** [cloudfoundry/php-buildpack#308]
**What version of Cloud Foundry and CF CLI are you using? (i.e. What is the output of running `cf curl /v2/info && cf version`?**

```
Unpacking cf-cli (6.45.0) ...
Setting up cf-cli (6.45.0) ...
api endpoint:   https://api.run.pivotal.io
api version:    2.136.0
```

**What version of the buildpack you are using?**

I haven't specified one.

**If you were attempting to accomplish a task, what was it you were attempting to do?**

I have a workflow setup on CircleCI which attempts to `cf push` my application (`symfony/demo`) to an app, then it should run composer install, followed by a few (standard) `post-install-cmd`s.

**What did you expect to happen?**

The app should be pushed, all installs run and then it's available.

**What was the actual behavior?**

The build fails due to an error in `post-install`.

Scripts expect my dependencies in `/tmp/app/`, but they are not.

Here is an excerpt, `composer install` runs, but `vendor/` is not in the output of the current directory (`ls -alh`):

```
...
     - Installing tgalopin/html-sanitizer (1.2.0): Loading from cache
     - Installing tgalopin/html-sanitizer-bundle (1.1.3): Loading from cache
     - Installing twig/extensions (v1.5.4): Loading from cache
   Generating autoload files
   ocramius/package-versions:  Generating version class...
   ocramius/package-versions: ...done generating version class
   > ls -lah
   total 6.4M
   drwxr-xr-x 18 vcap vcap 4.0K Jun 11 17:08 .
   drwxrwxrwt  1 root root  196 Jun 11 17:08 ..
   -rw-r--r--  1 vcap vcap 2.3K Jun 11 17:07 appveyor.yml
   drwxr-xr-x  4 vcap vcap   28 Jun 11 17:07 assets
   -rw-r--r--  1 vcap vcap  204 Jun 11 17:07 AUTH-dist
   drwxr-xr-x  2 vcap vcap   36 Jun 11 17:07 bin
   drwxr-xr-x  4 vcap vcap   29 Jun 11 17:07 .bp
   drwxr-xr-x  3 vcap vcap   39 Jun 11 17:07 .bp-config
   -rw-r--r--  1 vcap vcap 5.8M Jun 11 17:07 cf-cli_amd64.deb
   drwxr-xr-x  2 vcap vcap   24 Jun 11 17:07 .circleci
   -rw-r--r--  1 vcap vcap 2.7K Jun 11 17:07 composer.json
   -rw-r--r--  1 vcap vcap 248K Jun 11 17:07 composer.lock
   drwxr-xr-x  4 vcap vcap  116 Jun 11 17:07 config
   -rw-r--r--  1 vcap vcap  368 Jun 11 17:07 CONTRIBUTING.md
   drwxr-xr-x  2 vcap vcap   57 Jun 11 17:07 data
   -rw-r--r--  1 vcap vcap  131 Jun 11 17:07 .editorconfig
   -rw-r--r--  1 vcap vcap 1.2K Jun 11 17:07 .env
   -rw-r--r--  1 vcap vcap  177 Jun 11 17:07 .env.test
   drwxr-xr-x  8 vcap vcap   80 Jun 11 17:07 httpd
   drwxr-xr-x  3 vcap vcap   20 Jun 11 17:08 lib
   -rw-r--r--  1 vcap vcap 1.1K Jun 11 17:07 LICENSE
   drwxr-xr-x  2 vcap vcap    6 Jun 11 17:07 logs
   -rw-r--r--  1 vcap vcap  157 Jun 11 17:07 Makefile
   -rw-r--r--  1 vcap vcap  735 Jun 11 17:07 package.json
   drwxr-xr-x  9 vcap vcap   89 Jun 11 17:08 php
   -rw-r--r--  1 vcap vcap 1.4K Jun 11 17:07 .php_cs.dist
   -rw-r--r--  1 vcap vcap 1.2K Jun 11 17:07 phpunit.xml.dist
   drwxr-xr-x  3 vcap vcap  118 Jun 11 17:07 public
   -rw-r--r--  1 vcap vcap 1.5K Jun 11 17:07 README.md
   drwxr-xr-x 14 vcap vcap  217 Jun 11 17:07 src
   -rw-r--r--  1 vcap vcap  13K Jun 11 17:07 symfony.lock
   drwxr-xr-x 10 vcap vcap  136 Jun 11 17:07 templates
   drwxr-xr-x  6 vcap vcap   64 Jun 11 17:07 tests
   drwxr-xr-x  2 vcap vcap 4.0K Jun 11 17:07 translations
   -rw-r--r--  1 vcap vcap 1.6K Jun 11 17:07 .travis.yml
   -rw-r--r--  1 vcap vcap 1.2K Jun 11 17:07 webpack.config.js
   -rw-r--r--  1 vcap vcap 285K Jun 11 17:07 yarn.lock
Executing script assets:install --symlink --relative public [KO]
    [KO]
   Script assets:install --symlink --relative %PUBLIC_DIR% returned with error code 255
   !!  PHP Warning:  require(/tmp/app/vendor/autoload.php): failed to open stream: No such file or directory in /tmp/app/bin/console on line 11
   !!  PHP Fatal error:  require(): Failed opening required '/tmp/app/vendor/autoload.php' (include_path='../lib/php:/tmp/app/lib') in /tmp/app/bin/console on line 11
   !!  
   Script @auto-scripts was called via post-install-cmd
```

I'm debugging like so:

```
// composer.json
...
    "scripts": {
        "debug": [
            "pwd",
          	"ls -lah",
            "ls -lah vendor/",
			"@php -m"
		],
        "auto-scripts": {
            "assets:install --symlink --relative %PUBLIC_DIR%": "symfony-cmd",
            "security-checker security:check": "script"
        },
        "post-install-cmd": [
		    "@debug",
            "@auto-scripts"
        ],
        "post-update-cmd": [
            "@auto-scripts"
        ]
    },
...
```

**Steps to re-produce**

 1. clone [symfony/demo](http://github.com/symfony/demo)
 2. use the following manifest:
```
---
applications:
- name: redacted
  memory: 512M
  path: .
  buildpacks:
  - php_buildpack
```
 3. Setup `.bp-config/options.json `:
```
{
    "WEBDIR": "public"
}
```
 4. Example ~~`.circleci/config.yml`~~:
(not relevant, as a local `cf push` fails as well)
 5. ignore file: `.cfignore` 
```
composer.phar
vendor/
```

I think the issue is the path mangling, but I only recently started using this buildpack (and a little longer ago Cloud Foundry) and had no luck to figure this out. I see many related issues, but I don't see how they are solved.

I am not entirely sure why and where the libraries are installed or if there are special things to do when deploying a symfony application. Most applications expect vendor to be contained within the application code.

Please confirm where necessary:
* [x] I have included a log output
* [x] My log includes an error message
* [x] I have included steps for reproduction

**Comment by till** [comment]
Just realised, I forgot to include more log:

```
   > ls -lah vendor/
   ls: cannot access 'vendor/': No such file or directory
   Script ls -lah vendor/ handling the debug event returned with error code 2
   Script @debug was called via post-install-cmd
   -----> Composer command failed
   Traceback (most recent call last):
     File "/tmp/buildpacks/f4ef06d161ca990828fd4e1c6992e307/scripts/compile.py", line 59, in <module>
       .from_build_pack('lib/additional_commands')
     File "/tmp/buildpacks/f4ef06d161ca990828fd4e1c6992e307/lib/build_pack_utils/builder.py", line 212, in extensions
       process_extension(path, ctx, 'compile', process, args=[self])
     File "/tmp/buildpacks/f4ef06d161ca990828fd4e1c6992e307/lib/build_pack_utils/utils.py", line 69, in process_extension
       success(getattr(extn, to_call)(*args))
     File "/tmp/buildpacks/f4ef06d161ca990828fd4e1c6992e307/extensions/composer/extension.py", line 444, in compile
       return composer.compile(install)
     File "/tmp/buildpacks/f4ef06d161ca990828fd4e1c6992e307/lib/extension_helpers.py", line 154, in compile
       self._compile(install)
     File "/tmp/buildpacks/f4ef06d161ca990828fd4e1c6992e307/extensions/composer/extension.py", line 196, in _compile
       self.run()
     File "/tmp/buildpacks/f4ef06d161ca990828fd4e1c6992e307/extensions/composer/extension.py", line 343, in run
       *self._ctx['COMPOSER_INSTALL_OPTIONS'])
     File "/tmp/buildpacks/f4ef06d161ca990828fd4e1c6992e307/extensions/composer/extension.py", line 390, in run
       shell=True)
     File "/tmp/buildpacks/f4ef06d161ca990828fd4e1c6992e307/lib/build_pack_utils/runner.py", line 109, in stream_output
       raise CalledProcessError(retcode, cmd)
   build_pack_utils.runner.CalledProcessError: Command '<open file '<fdopen>', mode 'w' at 0x7f12cee84390>' returned non-zero exit status 2
   Failed to compile droplet: Failed to run finalize script: exit status 1
   Exit status 223
   Cell 9aa90abe-6a8f-4485-90d1-71da907de9a3 stopping instance 9d721944-63e7-445e-9de5-6386b53dbf1d
   Cell 9aa90abe-6a8f-4485-90d1-71da907de9a3 destroying container for instance 9d721944-63e7-445e-9de5-6386b53dbf1d
Error staging application: App staging failed in the buildpack compile phase
FAILED
Exited with code 1
```

**Comment by till** [comment]
When I `cf push` locally (same error):

```
   Executing script assets:install --symlink --relative public [KO]
    [KO]
   Script assets:install --symlink --relative %PUBLIC_DIR% returned with error code 255
   !!  PHP Warning:  require(/tmp/app/vendor/autoload.php): failed to open stream: No such file or directory in /tmp/app/bin/console on line 11
   !!  PHP Fatal error:  require(): Failed opening required '/tmp/app/vendor/autoload.php' (include_path='../lib/php:/tmp/app/lib') in /tmp/app/bin/console on line 11
   !!  
   Script @auto-scripts was called via post-install-cmd
   -----> Composer command failed
   Traceback (most recent call last):
     File "/tmp/buildpacks/f4ef06d161ca990828fd4e1c6992e307/scripts/compile.py", line 59, in <module>
       .from_build_pack('lib/additional_commands')
     File "/tmp/buildpacks/f4ef06d161ca990828fd4e1c6992e307/lib/build_pack_utils/builder.py", line 212, in extensions
       process_extension(path, ctx, 'compile', process, args=[self])
     File "/tmp/buildpacks/f4ef06d161ca990828fd4e1c6992e307/lib/build_pack_utils/utils.py", line 69, in process_extension
       success(getattr(extn, to_call)(*args))
     File "/tmp/buildpacks/f4ef06d161ca990828fd4e1c6992e307/extensions/composer/extension.py", line 444, in compile
       return composer.compile(install)
     File "/tmp/buildpacks/f4ef06d161ca990828fd4e1c6992e307/lib/extension_helpers.py", line 154, in compile
       self._compile(install)
     File "/tmp/buildpacks/f4ef06d161ca990828fd4e1c6992e307/extensions/composer/extension.py", line 196, in _compile
       self.run()
     File "/tmp/buildpacks/f4ef06d161ca990828fd4e1c6992e307/extensions/composer/extension.py", line 343, in run
       *self._ctx['COMPOSER_INSTALL_OPTIONS'])
     File "/tmp/buildpacks/f4ef06d161ca990828fd4e1c6992e307/extensions/composer/extension.py", line 390, in run
       shell=True)
     File "/tmp/buildpacks/f4ef06d161ca990828fd4e1c6992e307/lib/build_pack_utils/runner.py", line 109, in stream_output
       raise CalledProcessError(retcode, cmd)
   build_pack_utils.runner.CalledProcessError: Command '<open file '<fdopen>', mode 'w' at 0x7f4d04a13390>' returned non-zero exit status 255
   Failed to compile droplet: Failed to run finalize script: exit status 1
   Exit status 223
   Cell f237c36d-0a62-4579-a51b-b11ee1d58145 stopping instance e585380c-b5ea-4327-a3cc-4af35ff4a92c
   Cell f237c36d-0a62-4579-a51b-b11ee1d58145 destroying container for instance e585380c-b5ea-4327-a3cc-4af35ff4a92c
Error staging application: App staging failed in the buildpack compile phase
FAILED
```

TL;DR — the bit about CircleCI can be ignored too.

**Comment by till** [comment]
I figured out that I had to set the location of `vendor/` via `COMPOSER_VENDOR_DIR` (in `.bp-config/options.json`). I opened #309 to continue this discussion.
